compiler-core/src/ast/typed.rs

```diff
@@ -495,6 +556,112 @@ impl TypedExpr {
             _ => false,
         }
     }
+
+    pub fn to_string(&self) -> Option<EcoString> {
```

Comment:
Let's move this out into its own printer class and implement it by appending to a buffer to avoid all the extra allocations.